virtinst/guest.py

```diff
@@ -600,6 +600,17 @@ def enable_uefi(self):
         log.debug("Setting default UEFI path=%s", path)
         self.set_uefi_path(path)
 
+    def disable_uefi(self):
+        self.os.firmware = None
+        self.os.loader = None
+        self.os.loader_ro = None
+        self.os.loader_type = None
+        self.os.loader_secure = None
+        self.os.nvram = None
+        self.os.nvram_template = None
```

Comment:
Unfortunately this still doesn't work. Virt-install currently doesn't know about `format` attributes so it doesn't remove them and libvirt fails with this error:
```
ERROR    no loader path specified and firmware auto selection disabled
```

The --print-diff output looks like this with the current code:

```xml
--- Original XML
+++ Altered XML
@@ -9,14 +9,11 @@
   <memory unit="KiB">8388608</memory>
   <currentMemory unit="KiB">8388608</currentMemory>
   <vcpu placement="static">8</vcpu>
-  <os firmware="efi">
+  <os>
     <type arch="x86_64" machine="pc-q35-9.1">hvm</type>
-    <firmware>
-      <feature enabled="yes" name="enrolled-keys"/>
-      <feature enabled="yes" name="secure-boot"/>
-    </firmware>
-    <loader readonly="yes" secure="yes" type="pflash" format="qcow2">/usr/share/edk2/ovmf/OVMF_CODE_4M.secboot.qcow2</loader>
-    <nvram template="/usr/share/edk2/ovmf/OVMF_VARS_4M.secboot.qcow2" format="qcow2">/var/lib/libvirt/qemu/nvram/fedora40_VARS.qcow2</nvram>
+    <firmware/>
+    <loader format="qcow2"/>
+    <nvram format="qcow2"/>
     <boot dev="hd"/>
   </os>
   <features>
```

We need to add `format` into both `loader` and `nvram` or rework both to `XMLChildProperty`. Not sure if there is any other easy fix.

Reply:
aww fooey, was testing with an old VM without format='qcow2'. but this makes me realize it's hard to be future proof here. there's other bits we aren't covering like ./nvram/@type and ./loader/@stateless. Latest patch will clear all properties we are tracking, and then force removes the nvram, loader, and firmware XML nodes, so hopefully that covers us.